### PR TITLE
docs: add details about `images` options at `next.config.js` options

### LINKED ISSUE #62793: Docs: Please improve documentation for `next.config.js Options > images`
### What is the improvement or update you wish to see?

It should include `remotePatterns` and also specify what other values can `loader` have

### Is there any context that might help us understand?

`remotePatterns` is mentioned somewhere else in the documentation, so I went to the place where I expected a more in-depth description of that particular field and found nothing.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/api-reference/next-config-js/images

### PR DESCRIPTION
**Pull request by devjiwonchoi**
Since the [image configuration options](https://nextjs.org/docs/app/api-reference/components/image#configuration-options) have all the necessary options for `images` that need to be addressed in the [next.config.js options](https://nextjs.org/docs/app/api-reference/next-config-js/images), we refer to them without making significant changes.

Also, `use client` is required to use the `loader` option in the App Router but not Pages.

Closes #62793

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: c47514a00b8ddf95d1f230891ef612e2f0daceff

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: c47514a00b8ddf95d1f230891ef612e2f0daceff

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by styfle — CHANGES_REQUESTED** [review]
This seems to bury the lede.

We need to keep the code snippet.

**Comment by devjiwonchoi** [comment]
Thank you for reviewing!

File: docs/02-app/02-api-reference/01-components/image.mdx [hunk 1]
```diff
@@ -150,12 +152,34 @@ export default function Page() {
...
+      src="me.png"
+      alt="Picture of the author"
+      width={500}
+      height={500}
+    />
+  )
+}
+```
+
+</AppOnly>
```
Comment (devjiwonchoi) [inline docs/02-app/02-api-reference/01-components/image.mdx:181]:
```suggestion
</PagesOnly>
```

File: docs/02-app/02-api-reference/01-components/image.mdx [hunk 2]
```diff
@@ -474,21 +510,35 @@ If you want to use a cloud provider to optimize images instead of using the Next
 module.exports = {
   images: {
     loader: 'custom',
-    loaderFile: './my/image/loader.js',
+    loaderFile: './my/image/loader.mjs',
```
Comment (styfle) [inline docs/02-app/02-api-reference/01-components/image.mdx:513]:
```suggestion
    loaderFile: './my/image/loader.js',
```

File: docs/02-app/02-api-reference/01-components/image.mdx [hunk 3]
```diff
@@ -474,21 +510,35 @@ If you want to use a cloud provider to optimize images instead of using the Next
...
   },
 }
 ```
 
 This must point to a file relative to the root of your Next.js application. The file must export a default function that returns a string, for example:
 
-```js
+<AppOnly>
+
+```js filename="loader.mjs"
```
Comment (styfle) [inline docs/02-app/02-api-reference/01-components/image.mdx:522]:
```suggestion
```js filename="my/image/loader.js"
```

File: docs/02-app/02-api-reference/01-components/image.mdx [hunk 4]
```diff
@@ -474,21 +510,35 @@ If you want to use a cloud provider to optimize images instead of using the Next
...
 export default function myImageLoader({ src, width, quality }) {
   return `https://example.com/${src}?w=${width}&q=${quality || 75}`
 }
 ```
 
+</AppOnly>
+
+<PagesOnly>
+
+```js
```
Comment (styfle) [inline docs/02-app/02-api-reference/01-components/image.mdx:534]:
```suggestion
```js filename="my/image/loader.js"
```

File: docs/02-app/02-api-reference/05-next-config-js/images.mdx [hunk 5]
```diff
@@ -11,29 +11,43 @@ If you want to use a cloud provider to optimize images instead of using the Next
 module.exports = {
   images: {
     loader: 'custom',
-    loaderFile: './my/image/loader.js',
+    loaderFile: './my/image/loader.mjs',
```
Comment (styfle) [inline docs/02-app/02-api-reference/05-next-config-js/images.mdx:14]:
```suggestion
    loaderFile: './my/image/loader.js',
```

File: docs/02-app/02-api-reference/05-next-config-js/images.mdx [hunk 6]
```diff
@@ -11,29 +11,43 @@ If you want to use a cloud provider to optimize images instead of using the Next
...
   },
 }
 ```
 
 This `loaderFile` must point to a file relative to the root of your Next.js application. The file must export a default function that returns a string, for example:
 
-```js
+<AppOnly>
+
+```js filename="loader.mjs"
```
Comment (styfle) [inline docs/02-app/02-api-reference/05-next-config-js/images.mdx:23]:
```suggestion
```js filename="my/image/loader.js"
```

File: docs/02-app/02-api-reference/05-next-config-js/images.mdx [hunk 7]
```diff
@@ -11,29 +11,43 @@ If you want to use a cloud provider to optimize images instead of using the Next
...
 
+> **Good to know**: Customizing the image loader file, which accepts a function, requires using [Client Components](/docs/app/building-your-application/rendering/client-components) to serialize the provided function.
+
+To learn more about configuring the behavior of the built-in [Image Optimization API](/docs/app/building-your-application/optimizing/images) and the [Image Component](/docs/app/api-reference/components/image), see [Image Configuration Options](/docs/app/api-reference/components/image#configuration-options) for available options.
+
 </AppOnly>
 
 <PagesOnly>
 
+```js filename="loader.mjs"
```
Comment (styfle) [inline docs/02-app/02-api-reference/05-next-config-js/images.mdx:41]:
```suggestion
```js filename="my/image/loader.js"
```

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**